lib/logstash/inputs/udp.rb

```diff
@@ -97,10 +101,12 @@ def inputworker(number)
           decorate(event)
           event.set("host", client[3]) if event.get("host").nil?
           @output_queue.push(event)
+          metric.increment(:events)
```

Comment:
I've been thinking a little bit around, my thoughts:

### pros

* Code is DRY'ed.

### cons / open questions

* Add metrics attributes to codecs, as we only output plugin related metrics, I wonder if there is something else that `events` to be calculated there.
* What happens when a plugin is not using a codec? code gets duplicated then. See jdbc input case.
* What happens when we move to mills? 

This makes me think that even if the option is very nice and thoughtful, adding metrics to codecs will require a bit more thinking. Unless I'm missing something for sure.

Happy to agree on something with you PH.